build-logic/convention/src/main/kotlin/AndroidLibraryJacocoConventionPlugin.kt

```diff
@@ -23,13 +25,16 @@ import org.gradle.kotlin.dsl.getByType
 class AndroidLibraryJacocoConventionPlugin : Plugin<Project> {
     override fun apply(target: Project) {
         with(target) {
-            with(pluginManager) {
-                apply("org.gradle.jacoco")
-                apply("com.android.library")
+            pluginManager.apply("jacoco")
+            val androidExtension = extensions.getByType<LibraryExtension>()
+
+            androidExtension.buildTypes.configureEach {
+                // The jacoco plugin is applied automatically when any of these are set
```

Comment:
Is this true? I see that `pluginManager` is applying `jacoco` above directly still.

Reply:
It's complicated: you don't need to apply it if you only use the flags, but we're configuring `JacocoPluginExtension` in Jacoco.kt and that needs the plugin to be applied manually. I'll remove the comment, it's more confusing that helpful.